exporters/trace/elasticsearch/src/main/java/io/opencensus/exporter/trace/elasticsearch/JsonConversionUtils.java

```diff
@@ -0,0 +1,162 @@
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import javax.annotation.Nullable;
+
+/**
+ * Util to parse {@link SpanData} to json for {@link ElasticsearchTraceExporter}.
+ *
+ * @since 0.20.0
+ */
+public final class JsonConversionUtils {
+
+  private static final String ELASTICSEARCH_DATE_PATTERN = "yyyy-MM-dd'T'HH:mm:ss.SSSZZ";
+
+  private JsonConversionUtils() {}
+
+  private static String encodeTraceId(TraceId traceId) {
+    return traceId.toLowerBase16();
+  }
+
+  private static String encodeSpanId(SpanId spanId) {
+    return spanId.toLowerBase16();
+  }
+
+  private static String toSpanName(SpanData spanData) {
+    return spanData.getName();
+  }
+
+  private static long toMillis(Timestamp timestamp) {
+    return SECONDS.toMillis(timestamp.getSeconds()) + NANOSECONDS.toMillis(timestamp.getNanos());
+  }
+
+  private static long toMillis(Timestamp start, Timestamp end) {
+    Duration duration = end.subtractTimestamp(start);
+    return duration.toMillis();
+  }
+
+  private static Date toDate(Timestamp timestamp) {
+    return new Date(toMillis(timestamp));
+  }
+
+  private static String formatDate(Timestamp timestamp) {
+    return new SimpleDateFormat(ELASTICSEARCH_DATE_PATTERN).format(toDate(timestamp));
+  }
+
+  @Nullable
+  @SuppressWarnings("nullness")
+  private static String attributeValueToString(AttributeValue attributeValue) {
+    return attributeValue.match(
+        Functions.returnToString(),
+        Functions.returnToString(),
+        Functions.returnToString(),
+        Functions.returnToString(),
+        Functions.returnConstant(""));
+  }
+
+  private static String toSpanType(SpanData spanData) {
+    if (spanData.getParentSpanId() == null || Boolean.TRUE.equals(spanData.getHasRemoteParent())) {
+      return "ENTRY";
+    }
+    if (spanData.getName().startsWith("Sent.")) {
+      return "EXIT";
+    }
+    return "INTERMEDIATE";
+  }
+
+  /**
+   * Converts a collection of {@link SpanData} to a Collection of json string.
+   *
+   * @param appName the name of app to include in traces.
+   * @param spanDataList Collection of {@code SpanData} to be converted to json.
+   * @return Collection of {@code SpanData} converted to JSON to be indexed.
+   */
+  public static List<String> convertToJson(String appName, Collection<SpanData> spanDataList) {
+    List<String> spanJson = new ArrayList<String>();
+    if (spanDataList == null) {
+      return spanJson;
+    }
+    StringBuilder sb = new StringBuilder();
+    for (final SpanData span : spanDataList) {
+      final SpanContext spanContext = span.getContext();
+      final SpanId parentSpanId = span.getParentSpanId();
+      final Timestamp startTimestamp = span.getStartTimestamp();
+      final Timestamp endTimestamp = span.getEndTimestamp();
+      final Status status = span.getStatus();
+      if (status == null || endTimestamp == null) {
```

Comment:
Null `status` means OK (https://github.com/census-instrumentation/opencensus-proto/issues/158). Do you only want the error spans?